src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -2006,11 +2006,11 @@ NOTE: if you're sure you want to do this, please open an issue as to why. In the
             // Tests that use compiler libraries may inherit the `-lLLVM` link
             // requirement, but the `-L` library path is not propagated across
             // separate compilations. We can add LLVM's library path to the
-            // platform-specific environment variable as a workaround.
+            // rustc args as a workaround.
             if !builder.config.dry_run() && suite.ends_with("fulldeps") {
                 let llvm_libdir =
                     command(&llvm_config).arg("--libdir").run_capture_stdout(builder).stdout();
-                add_link_lib_path(vec![llvm_libdir.trim().into()], &mut cmd);
+                cmd.env("RUSTFLAGS", format!("-Clink-arg=-L{llvm_libdir}"));
```

Comment:
this will overwrite any existing RUSTFLAGS. see if there’s a `cargo.rustflag` function you can use instead; if not there should be an existing one somewhere you can copy from.

Reply:
also, this will break on windows msvc.

Reply:
I wasn't able to find a rustflags helper in scope for this function, so I did it manually. I also now check target.is_msvc() in order to determined which style of linker flag to emit.